### DISCUSSION TITLE
Refiners with "Microsoft Search" data source

### DISCUSSION THREAD

**Discussion by simesy** [microsoft-search/pnp-modern-search#2034] [Q&A] [answered]
I've been using refiners with the SharePoint Search data source. Can someone explain why there is no refiners property for the Microsoft Search data source?

**Comment by wobba** [comment]
You can use the filter web part with the Microsoft Search data source as well. What's your issue?

**Reply by simesy** [reply]
I'm using the "Refinement filters" property a lot, to create specialised search solutions, so just wondering what is the correct way to do this with Microsoft Search data source which doesn't have the "Refinement filters" property.

<img width="421" alt="Screen Shot 2022-04-26 at 9 24 23 am" src="https://user-images.githubusercontent.com/188856/165190378-8aa0e0f1-176d-40c7-8b3d-74444783239e.png">

<img width="424" alt="Screen Shot 2022-04-26 at 9 22 38 am" src="https://user-images.githubusercontent.com/188856/165190392-aafbe342-3ea1-4cc0-9b9c-58de935b77de.png">

**Comment by edarroudi** [comment]
You can use the Query template like this: 
![image](https://user-images.githubusercontent.com/935493/165244294-4c559fa4-400b-4f92-b93a-528be14836de.png)

Attention:
QueryTemplate is right now used to set the queryString. MS is adding a  "queryTemplate" node to the request as well (BETA). But it shouldn't matter in your case.

**Reply by simesy** [reply]
This box appears for the SharePoint Search source but not for the Microsoft Search source. I wonder if something is different for me... I'm using the latest PnP Search, or so I thought, Version: 4.4.0.

Edit: sh1t why is my version old again... I seem to be getting the global version and not the one my local Apps Catalog.

**Comment by simesy** [comment, marked as answer]
Turns out the problem is my site is inheriting the global version which is old. I tested 4.6.1 on my dev tenant and the properties appear as discussed here. Sorry for the noise.
